src/System.Management.Automation/engine/regex.cs

```diff
@@ -185,15 +185,16 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
 
             if (pattern == null)
             {
-                throw PSTraceSource.NewArgumentNullException("pattern");
+                throw PSTraceSource.NewArgumentNullException(nameof(pattern));
             }
 
             if (charsNotToEscape == null)
             {
-                throw PSTraceSource.NewArgumentNullException("charsNotToEscape");
+                throw PSTraceSource.NewArgumentNullException(nameof(charsNotToEscape));
             }
 
-            char[] temp = new char[pattern.Length * 2 + 1];
+            const int StackAllocThreshold = 256;
```

Comment:
Why 256? How do you decide on the threshold?

Reply:
Comment added: The size is less than MaxShortPath = 260